firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/common/FirebaseCrashlyticsReportManagerTest.java

```diff
@@ -156,19 +167,123 @@ public void testOnFatalEvent_addsLogsToEvent() {
+
+    final CustomAttribute customAttribute1 =
+        CustomAttribute.builder().setKey(testKey1).setValue(testValue1).build();
+    final CustomAttribute customAttribute2 =
+        CustomAttribute.builder().setKey(testKey2).setValue(testValue2).build();
+
+    final ImmutableList<CustomAttribute> expectedCustomAttributes =
+        ImmutableList.from(customAttribute1, customAttribute2);
+
+    when(reportMetadata.getCustomKeys()).thenReturn(attributes);
+
+    reportManager.onBeginSession("testSessionId");
+    reportManager.onNonFatalEvent(mockException, mockThread);
+
+    verify(mockEventAppBuilder).setCustomAttributes(expectedCustomAttributes);
+    verify(mockEventAppBuilder).build();
+    verify(mockEventBuilder).setApp(mockEventApp);
+    verify(mockEventBuilder).build();
+    verify(logFileManager).clearLog();
+  }
+
+  @Test
+  public void testOnNonFatalEvent_addsNoKeysToEventWhenNoneAvailable() {
+    mockEventInteractions(System.currentTimeMillis());
+
+    final Map<String, String> attributes = new HashMap<>();
+
+    when(reportMetadata.getCustomKeys()).thenReturn(attributes);
+
+    reportManager.onBeginSession("testSessionId");
+    reportManager.onNonFatalEvent(mockException, mockThread);
+
+    verify(mockEventAppBuilder, never()).setCustomAttributes(any());
+    verify(mockEventAppBuilder, never()).build();
+    verify(mockEventBuilder, never()).setApp(mockEventApp);
+    verify(mockEventBuilder).build();
+    verify(logFileManager).clearLog();
+  }
+
+  @Test
+  public void testOnFatalEvent_addsSortedKeysToEvent() {
+    mockEventInteractions(System.currentTimeMillis());
+
+    final String testKey1 = "testKey1";
+    final String testValue1 = "testValue1";
+    final String testKey2 = "testKey2";
+    final String testValue2 = "testValue2";
+
+    final Map<String, String> attributes = new HashMap<>();
+    attributes.put(testKey1, testValue1);
+    attributes.put(testKey2, testValue2);
+
+    final CustomAttribute customAttribute1 =
+        CustomAttribute.builder().setKey(testKey1).setValue(testValue1).build();
+    final CustomAttribute customAttribute2 =
+        CustomAttribute.builder().setKey(testKey2).setValue(testValue2).build();
+
+    final ImmutableList<CustomAttribute> expectedCustomAttributes =
+        ImmutableList.from(customAttribute1, customAttribute2);
+
+    when(reportMetadata.getCustomKeys()).thenReturn(attributes);
+
+    reportManager.onBeginSession("testSessionId");
+    reportManager.onFatalEvent(mockException, mockThread);
+
+    verify(mockEventAppBuilder).setCustomAttributes(expectedCustomAttributes);
+    verify(mockEventAppBuilder).build();
+    verify(mockEventBuilder).setApp(mockEventApp);
+    verify(mockEventBuilder).build();
+    verify(logFileManager).clearLog();
+  }
+
+  @Test
+  public void testOnFatalEvent_addsNoKeysToEventWhenNoneAvailable() {
+    mockEventInteractions(System.currentTimeMillis());
+
+    final Map<String, String> attributes = new HashMap<>();
```

Comment:
How do you feel about `Collections.emptyMap()` (or `ImmutableMap.of()`)

Reply:
:+1: